src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -66,24 +65,29 @@ public class TestWorker
         /// <param name="apartmentState">The apartment state to use for running tests</param>
         public TestWorker(WorkItemQueue queue, string name, ApartmentState apartmentState)
         {
-            _readyQueue = queue;
-
-            _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
-            _workerThread.Name = name;
-            _workerThread.SetApartmentState(apartmentState);
+            WorkQueue = queue;
+            Name = name;
+            Apartment = apartmentState;
```

Comment:
Ah ok! I was looking for some significance in why the thread creation needed to be moved. If it's just a minor optimisation, that makes sense. 😄